modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/DiscoveryClientSocketTest.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.spi.discovery.tcp;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.ServerSocket;
+import java.net.Socket;
+import javax.net.SocketFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLServerSocketFactory;
+import org.apache.ignite.internal.IgniteFutureTimeoutCheckedException;
+import org.apache.ignite.internal.IgniteInternalFuture;
+import org.apache.ignite.internal.util.IgniteUtils;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.apache.ignite.ssl.SslContextFactory;
+import org.apache.ignite.testframework.GridTestUtils;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+/**
+ * Check a ssl socket configuration which used in discovery.
+ */
+public class DiscoveryClientSocketTest {
+    /** Port to listen. */
+    public static final int PORT_TO_LNSR = 12346;
+    /** Host for bind. */
+    public static final String HOST = "localhost";
+
+    /** SSL server socket factory. */
+    private SSLServerSocketFactory sslSrvSockFactory;
+    /** SSL socket factory. */
+    private SocketFactory sslSockFactory1;
+    /** Fake TCP discovery SPI. */
+    private TcpDiscoverySpi fakeTcpDiscoverySpi;
+
+    /**
+     * Configure SSL and Discovery.
+     */
+    @Before
+    public void before() {
+        SslContextFactory socketFactory = (SslContextFactory)GridTestUtils.sslTrustedFactory("node01", "trustone");
+        SSLContext sslCtx = socketFactory.create();
+
+        sslSrvSockFactory = sslCtx.getServerSocketFactory();
+        sslSockFactory1 = sslCtx.getSocketFactory();
+        fakeTcpDiscoverySpi = new TcpDiscoverySpi();
+
+        fakeTcpDiscoverySpi.setSoLinger(1);
+    }
+
+    /**
+     * It creates a SSL socket server and client for checks correctness closing when write exceed read.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void sslSocketTest() throws Exception {
```

Comment:
If I comment out a line , the test will not fall. 
TcpDiscoverySpi#configureSocketOptions
socket.setSoLinger(get So Linger() >= 0, get So Linger());

Reply:
I slightly modified the test, now it is going with timeout, even if anyone removed SoLinger.